core/templates/components/graph-services/graph-layout.service.ts

```diff
@@ -547,64 +548,65 @@ export class StateGraphLayoutService {
         var targetWidth = link.target.width;
         var targetHeight = link.target.height;
 
-        var dx = targetx - sourcex;
-        var dy = targety - sourcey;
-
-        /* Fractional amount of truncation to be applied to the end of
-           each link. */
-        var startCutoff = (sourceWidth / 2) / Math.abs(dx);
-        var endCutoff = (targetWidth / 2) / Math.abs(dx);
-        if (dx === 0 || dy !== 0) {
-          startCutoff = (
-            (dx === 0) ? (sourceHeight / 2) / Math.abs(dy) :
-            Math.min(startCutoff, (sourceHeight / 2) / Math.abs(dy)));
-          endCutoff = (
-            (dx === 0) ? (targetHeight / 2) / Math.abs(dy) :
-            Math.min(endCutoff, (targetHeight / 2) / Math.abs(dy)));
-        }
+        if (targetx && sourcex && targety && sourcey && targetWidth &&
+          sourceWidth && targetHeight && sourceHeight) {
```

Comment:
Why is this needed?

Reply:
The best approach would again be to get rid of the `null`s, or at least make only one of the values indicate the the rest will be `null`. But if we keep the if it should look like this.
```suggestion
        if (
          target !== null && 
          sourcex !== null && 
          targety !== null && 
          sourcey !== null &&
          targetWidth  !== null && 
          sourceWidth !== null && 
          targetHeight !== null && 
          sourceHeight !== null 
        ) {
```